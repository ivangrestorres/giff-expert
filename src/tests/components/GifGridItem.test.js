import React from "react";
const { shallow } = require("enzyme");
const { GifGridItem } = require("../../components/GifGridItem");

describe("Pruebas <GifGridItem />", () => {
    const title = "Titulo";
    const url = "https://localhost/algo.jpg";
    const wrapper = shallow(<GifGridItem title={title} url={url} />);
    test("debe mostrar el componente correctamente ", () => {
        expect(wrapper).toMatchSnapshot();
    });
    test("debe tener un parrafo con el titulo", () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });

    test("debe tener la imagen igual al URL y Alt de los props", () => {
        const img = wrapper.find("img");
        const { src, alt } = img.props();

        expect(alt).toBe(title);
        expect(src).toBe(url);
    });

    test("debe tener animate__fadeIn", () => {
        const div = wrapper.find("div");
        const { className } = div.props();

        console.log(className.includes("animate__fadeIn"));

        expect(className.includes("animate__fadeIn")).toBe(true);
    });
});
