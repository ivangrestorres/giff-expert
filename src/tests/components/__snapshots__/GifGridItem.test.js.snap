// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas <GifGridItem /> debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Titulo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Titulo
  </p>
</div>
`;
